harness.tf

```diff
@@ -0,0 +1,24 @@
+module "delegate" {
+  count = var.harness_delegate == true ? 1 : 0
+
+  source  = "./harness"
+
+  account_id       = jsondecode(data.vault_kv_secret_v2.delegate_secrets.data_json)["DELEGATE_ACCOUNT_ID"]
+  delegate_token   = jsondecode(data.vault_kv_secret_v2.delegate_secrets.data_json)["DELEGATE_TOKEN"]
+  delegate_name    = "${var.cluster_name}-harness-delegate"
+  namespace        = "harness-delegate-ng"
+  manager_endpoint = "https://app.harness.io/gratis"
```

Comment:
does this mean every cluster now will be tied to an external harness service ?

Reply:
From the description: "Harness Delegate is a service you run in your local network or VPC to connect your artifacts, infrastructure, collaboration, verification, and other providers, with Harness Manager. The first time you connect Harness to a third-party resource, Harness Delegate is installed in your target infrastructure, for example, a Kubernetes cluster. After the delegate is installed, you connect to third-party resources. The delegate performs all operations, including deployment and integration."

I have real concerns about adding this to our infrastructure.

Reply:
Right now harness is just for CI/CD and release automation. we don't expect it to be installed or used in customer clusters. It should only be used for testing/feature clusters, dev-ci, and staging.

Reply:
what about Azure ?

Reply:
I just want it to be crystal clear here, this cannot go into Customer Clusters, ever.   We have many customers that either block ingress or even more would never allow an external to connect to a service with an RBAC of cluster-admin.

Reply:
Suggest you enhance the count expression to include that the account must include "indico" in it.

Reply:
i'm assuming the var `aws_account`?

Reply:
Are you ignoring Azure?

Reply:
I believe we'll get to supporting multiple clouds with the cluster creation configurations work that involves coming up with a list of supported configurations (cloud, feature flag combos, etc).